### ISSUE TITLE
Execution failed for task ':moor_ffi:extractSqlite'.

### ISSUE THREAD

**Issue by Mike278** [simolus3/moor#282]
After upgrading to the latest Flutter/Dart, I'm seeing this build failure: 
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':moor_ffi:extractSqlite'.
> Cannot expand ZIP '/home/mike/.pub-cache/hosted/pub.dartlang.org/moor_ffi-0.2.0/android/sqlite_3290000.zip' as it does not exist.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 16s
Finished with error: Gradle task assembleDebug failed with exit code 1
```

Flutter version:
```
Flutter 1.12.13+hotfix.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 27321ebbad (7 days ago) • 2019-12-10 18:15:01 -0800
Engine • revision 2994f7e1e6
Tools • Dart 2.7.0
```

Resolved moor versions:
```
moor:file:///home/mike/.pub-cache/hosted/pub.dartlang.org/moor-2.1.1/lib/
moor_ffi:file:///home/mike/.pub-cache/hosted/pub.dartlang.org/moor_ffi-0.2.0/lib/
moor_flutter:file:///home/mike/.pub-cache/hosted/pub.dartlang.org/moor_flutter-2.0.0/lib/
moor_generator:file:///home/mike/.pub-cache/hosted/pub.dartlang.org/moor_generator-2.1.1/lib/
```

Deleting the `.pub-cache/hosted/pub.dartlang.org` dirs to force a redownload doesn't seem to work.

**Comment by simolus3** [comment]
In `moor_ffi 0.2.0`, the idea is that we download sqlite on the first build. That's a bad idea for multiple reasons, and we won't do that in the next version.

You can keep the stable moor version and fix this problem by only overriding `moor_ffi`:
```yaml
  moor_ffi:
    git:
      url: https://github.com/simolus3/moor.git
      ref: develop
      path: moor_ffi
```

First compilation might take a while when bundling an apk, we compile sqlite with `-O3` starting from that version. I'll try to release that version tomorrow or so.

**Comment by Mike278** [comment]
Ok sounds good. Is there something I can do to force the build to download sqlite with `moor_ffi 0.2.0`, or is that release just broken? I'm surprised [this](https://github.com/simolus3/moor/blob/master/moor_ffi/android/build.gradle#L58) doesn't work, it seems like it should.

**Comment by simolus3** [comment]
> did that just use the platform's sqlite installation?

As far as I remember we always downloaded sqlite on Android. We can't use the native sqlite3.so on Android 6 and above.

> I'm surprised this doesn't work, it seems like it should.

Yeah it's pretty weird that the script just stopped working with a newer Flutter version - I couldn't even reproduce the failure you were seeing because I ran into #256 with a new project.

> build to download sqlite with `moor_ffi 0.2.0`

You can probably chdir into the `moor_ffi/android` folder in `.pub_cache` and run `gradle extractSqlite` in there. If everything fails, you could also download the latest amalgamation [here](https://www.sqlite.org/download.html) and extract the `sqlite3.c` file into `moor_ffi/android/cpp/sqlite3.c`. I'll try to get the 0.3.0 release ready now.

**Comment by simolus3** [comment]
Should be fixed in version 0.3.0 of `moor_ffi`

**Comment by Mike278** [comment]
Yep working perfectly, thanks so much for your help as always!
